### PR TITLE
Add default values to `Admin::EditAuctionViewModel`

### PR DESCRIPTION
**Pull request by jessieay**
* Started_at and ended_at are required, but `delivery_deadline` input
  (which is on create only) is not, leaving us with a blank
  `delivery_due_at` field on edit
* Was receiving errors on staging (`undefined method strftime for nil`)
* Also update `auctions#new` defaults to be ET defaults suggested in #560

### CODE REVIEW

File: app/view_models/admin/edit_auction_view_model.rb [hunk 1]
```diff
@@ -14,24 +14,44 @@ def new_record?
   end
 
   def hour_default(field)
-    dc_time(field).strftime("%l").strip
+    if field_value(field).present?
+      dc_time(field).strftime("%l").strip
+    else
+      "1"
```
Comment (harrisj) [inline app/view_models/admin/edit_auction_view_model.rb]: I almost wonder if it makes sense to just call the NewAuctionViewModel for these, but I suppose a little duplication is fine too

File: app/view_models/admin/edit_auction_view_model.rb [hunk 2]
```diff
@@ -14,24 +14,44 @@ def new_record?
...
+      dc_time(field).to_date
+    else
+      DcTimePresenter.convert(Date.today).to_date
+    end
   end
 
   private
 
   def dc_time(field)
     DcTimePresenter.convert(auction.send("#{field}_at"))
```
Comment (harrisj) [inline app/view_models/admin/edit_auction_view_model.rb]: Can you have this call field_value too?

File: app/view_models/admin/new_auction_view_model.rb [hunk 3]
```diff
@@ -20,6 +20,6 @@ def meridiem_default(_field)
   end
 
   def date_default(_field)
-    Date.today
+    DcTimePresenter.convert(Date.today).to_date
```
Comment (harrisj) [inline app/view_models/admin/new_auction_view_model.rb]: Good catch. This would be an issue late at night.

File: spec/view_models/admin/edit_auction_view_model_spec.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
+require 'rails_helper'
+
+describe Admin::EditAuctionViewModel do
```
Comment (harrisj) [inline spec/view_models/admin/edit_auction_view_model_spec.rb]: Oh man, did we really have no tests? This is why I want to figure out the issues with coverage I guess!

File: app/view_models/admin/edit_auction_view_model.rb (hunk 1 above)
Comment (jessieay) [inline app/view_models/admin/edit_auction_view_model.rb]: yeah let me think about how we can make this less ugly

File: app/view_models/admin/edit_auction_view_model.rb (hunk 2 above)
Comment (jessieay) [inline app/view_models/admin/edit_auction_view_model.rb]: yeah good idea that will clean this up a bit

File: app/view_models/admin/edit_auction_view_model.rb (hunk 1 above)
Comment (harrisj) [inline app/view_models/admin/edit_auction_view_model.rb]: I wouldn't sweat it too much though.

Sent from my iPhone

On May 27, 2016, at 6:20 PM, Jessie A. Young <notifications@github.com>
wrote:

In app/view_models/admin/edit_auction_view_model.rb
<https://github.com/18F/micropurchase/pull/620#discussion_r64973576>:

> @@ -14,24 +14,44 @@ def new_record?
>    end
>
>    def hour_default(field)
> -    dc_time(field).strftime("%l").strip
> +    if field_value(field).present?
> +      dc_time(field).strftime("%l").strip
> +    else
> +      "1"

yeah let me think about how we can make this less ugly

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/18F/micropurchase/pull/620/files/a0f13b309952c5e916ff6c0fa72877d75b258af9#r64973576>,
or mute the thread
<https://github.com/notifications/unsubscribe/AAAH_DC0bof-J5z-cT8wA5FY_J41Ymhwks5qF25HgaJpZM4Io3WW>
.

File: spec/view_models/admin/edit_auction_view_model_spec.rb (hunk 4 above)
Comment (jessieay) [inline spec/view_models/admin/edit_auction_view_model_spec.rb]: these are tested at the integration level but we didn't have anything that hit these edge case so thought I'd throw in some unit tests!

**Comment by jessieay** [comment]
@harrisj updates made at de9aefe6034a16a82e524f692da1b46b446a99e6 -- let me know what you think!

**Comment by harrisj** [comment]
Looks good. Let's merge
